src/mam/mod_mam_muc.erl

```diff
@@ -766,57 +763,9 @@ report_issue(Reason, Stacktrace, Issue, #jid{lserver = LServer, luser = LUser},
     ?ERROR_MSG("issue=~p, server=~p, user=~p, reason=~p, iq=~p, stacktrace=~p",
                [Issue, LServer, LUser, Reason, IQ, Stacktrace]).
 
-
-%% ----------------------------------------------------------------------
-%% Dynamic params module
-
-%% compile_params_module([
-%%      {add_archived_element, boolean()}
-%%      ])
-compile_params_module(Params) ->
-    CodeStr = params_helper(Params),
-    {Mod, Code} = dynamic_compile:from_string(CodeStr),
-    code:load_binary(Mod, "mod_mam_muc_params.erl", Code).
-
-params_helper(Params) ->
-    %% Try is_complete_message opt for backwards compatibility
-    {IsArchivableModule, IsArchivableFunction} =
-        case proplists:get_value(is_archivable_message, Params) of
-            undefined ->
-                case proplists:get_value(is_complete_message, Params) of
-                    undefined -> {mod_mam_utils, is_archivable_message};
-                    OldStyleMod -> {OldStyleMod, is_complete_message}
-                end;
-
-            Mod -> {Mod, is_archivable_message}
-        end,
-
-    Format =
-        io_lib:format(
-          "-module(mod_mam_muc_params).~n"
-          "-compile(export_all).~n"
-          "add_archived_element() -> ~p.~n"
-          "add_stanzaid_element() -> not ~p.~n"
-          "is_archivable_message(Mod, Dir, Packet) -> ~p:~p(Mod, Dir, Packet, false).~n" %false as we don't support chat markers for groupchats
-          "default_result_limit() -> ~p.~n"
-          "max_result_limit() -> ~p.~n"
-          "params() -> ~p.~n",
-          [proplists:get_bool(add_archived_element, Params),
-           proplists:get_bool(no_stanzaid_element, Params),
-           IsArchivableModule, IsArchivableFunction,
-	   proplists:get_value(default_result_limit, Params, 50),
-	   proplists:get_value(max_result_limit, Params, 50),
-           Params]),
-    binary_to_list(iolist_to_binary(Format)).
-
-set_params(Params) ->
-    compile_params_module(Params).
-
-%% @doc Enable support for `<archived/>' element from MAM v0.2
--spec add_archived_element() -> boolean().
-add_archived_element() ->
-    mod_mam_muc_params:add_archived_element().
-
--spec add_stanzaid_element() -> boolean().
-add_stanzaid_element() ->
-    mod_mam_muc_params:add_stanzaid_element().
+-spec is_archivable_message(Host :: ejabberd:lserver(), Dir :: incoming | outgoing,
+                            Packet :: exml:element()) -> boolean().
+is_archivable_message(Host, Dir, Packet) ->
```

Comment:
Maybe it should be extracted to `mod_mam_utils`? I believe it is a duplicate from `mod_mam`.

Reply:
@arcusfelis wanted to have param-driven functions out of `mod_mam_utils` and I sort-of agree. Choosing parameters went to `mod_mam_params`, actual logic went to `mod_mam_utils` and a convenience shim connecting them went to mam modules. Duplication is a pity here, but it stems more from duplication of other parts of `mod_mam`/`mod_mam_muc`.

Reply:
OK, I understand. May be a subject of refactoring in the future then, maybe there will be a dedicated module for param-driven MAM utils.